src/runtime-rs/crates/hypervisor/src/qemu/qmp.rs

```diff
@@ -128,6 +142,153 @@ impl Qmp {
 
         Ok(hotunplugged)
     }
+
+    pub fn set_guest_memory_block_size(&mut self, size: u64) {
+        self.guest_memory_block_size = size;
+    }
+
+    pub fn guest_memory_block_size(&self) -> u64 {
+        self.guest_memory_block_size
+    }
+
+    pub fn hotplugged_memory_size(&mut self) -> Result<u64> {
+        let memory_frontends = self.qmp.execute(&qapi_qmp::query_memory_devices {})?;
+
+        let mut hotplugged_mem_size = 0_u64;
+
+        info!(sl!(), "hotplugged_memory_size(): iterating over dimms");
+        for mem_frontend in &memory_frontends {
+            if let qapi_qmp::MemoryDeviceInfo::dimm(dimm_info) = mem_frontend {
+                let id = match dimm_info.data.id {
+                    Some(ref id) => id.clone(),
+                    None => "".to_owned(),
+                };
+
+                info!(
+                    sl!(),
+                    "dimm id: {} size={}, hotplugged: {}",
+                    id,
+                    dimm_info.data.size,
+                    dimm_info.data.hotplugged
+                );
+
+                if dimm_info.data.hotpluggable && dimm_info.data.hotplugged {
+                    hotplugged_mem_size += dimm_info.data.size as u64;
+                }
+            }
+        }
+
+        Ok(hotplugged_mem_size)
+    }
+
+    pub fn hotplug_memory(&mut self, size: u64) -> Result<()> {
+        let memdev_idx = self
```

Comment:
What about renaming it to `hotplugged_memdev_count `? I know it will be used to construct a new id below, but the code itself ends with `count()`. 😉

Reply:
Yeah, well, conceptually the number starts as a count and ends as an index, so much is sure.  Now the question is where exactly the transition happens.  I put it where I did I guess precisely because the line ends in `count()` but stores in `idx` - I kind of like how explicit the transition in the number's meaning is this way. :-)

The alternative would be calling the variable `memdev_count` and then have
```
let memory_backend_id = format!("hotplugged-{}", memdev_count);
```
Is that better though?

Reply:
Yeah, it was just an opinion. I will follow your decision. At least, I was confused when I firstly read this line and got it clear at the next line. Let's see what the 2nd reviewer would view this.

Reply:
Thanks.  There will always be some friction I suspect since fundamentally, we're taking a count and casting it into an index.

Reply:
It is safe to assume that existing ids are < `count()` since the indexing is part of a naming scheme that's internal to the driver and owned by it (that's the idea anyway :-)).

As for not maintaining it on the shim side, this is actually part of a conscious decision to treat QEMU as an authoritative source of information.  Considering that QEMU keeps track of this and also gives us a convenient mechanism to query it, it doesn't seem beneficial (to me anyway) to write & maintain extra code in the runtime to track the very same information, even introducing a danger of going out of sync (say due to a bug in the runtime-side accounting).

Reply:
Fair enough.